### PR TITLE
Fix #305 and #306: read-for-update operations use primary

### LINKED ISSUE #305: Are lock operations done w/ primary read preference?
For example, does ping always check primary?

I looked through code, it looks like it is not set, so should default to primary (says mongo docs), but I'm not sure if that default is overridden anywhere.

I ask because I occasionally see some weird things like, a client will acquire a lock successfully, then ping it, and now they get an invalid lock exception just a half a second later, and TTL is 90 seconds.

### PR DESCRIPTION
**Pull request by bserdar**

### CODE REVIEW

**Comment by alechenninger** [comment]
Is there any way to test this?

**Comment by bserdar** [comment]
If this can be reproduced in dev env, we can push it, and try to reproduce
it.

**Comment by alechenninger** [comment]
Sorry I mean, automated unit or integration test of some kind. I don't think it's a blocker to merge, but it might be good to have a test harness with artificially broken / slow mongo replication that we could use to test some of these weird edge cases.

**Comment by bserdar** [comment]
That's pretty difficult to setup for a unit test. Let me think about it/

File: mongo/src/main/java/com/redhat/lightblue/mongo/crud/IterateAndUpdate.java [hunk 1]
```diff
@@ -123,6 +124,8 @@ public void update(CRUDOperationContext ctx,
             measure.begin("iteration");
             int batchStartIndex=0; // docUpdateAttempts[batchStartIndex] is the first doc in this batch
             while (cursor.hasNext()) {
+                // Read from primary for read-for-update operations
+                cursor.setReadPreference(ReadPreference.primary());
```
Comment (alechenninger) [inline mongo/src/main/java/com/redhat/lightblue/mongo/crud/IterateAndUpdate.java]: I think this needs to be set before the call to hasNext -- at that point we have already executed the query right?

Comment (bserdar) [inline mongo/src/main/java/com/redhat/lightblue/mongo/crud/IterateAndUpdate.java]: fixed.
